scripts/start/prepare_and_run.R

```diff
@@ -884,89 +887,80 @@ run <- function(start_subsequent_runs = TRUE) {
-             }
-          }
-       }
-       
-       save(cfg, file = RData_file)
-
-       # 3. ... use it as 'input_ref.gdx' and start these runs (known as subsequent runs).
-       if (cfg_main$RunsUsingTHISgdxAsInput[run,,drop=FALSE]$path_gdx_ref == cfg_main$title) {
-          load(RData_file)
-          cat("Writing the path for input_ref.gdx to ",RData_file,"\n")
-          # ...change the path_gdx_ref field of the subsequent run to the fulldata gdx of the current run ...
-          cfg$files2export$start['input_ref.gdx'] <- paste0(cfg_main$remind_folder,"/",cfg_main$results_folder,"/fulldata.gdx")
-          save(cfg, file = RData_file)
-          # Subsequent runs will be started in submit.R using the RData files written above after the current run has finished.
-          cat("Starting subsequent run ",run,"\n")
-          source("scripts/start/submit.R")
-          submit(cfg)
-       } else {
-          cat('\nNo subsequent run was set for this scenario\n')
-       }
-       
-    }
+      message("\nPrepare subsequent run ", run, ":")
+      RData_file <- paste0(run,".RData")
+      load(RData_file)
+
+      pathes_to_gdx <- intersect(possible_pathes_to_gdx, names(cfg$files2export$start))
+
+      gdx_na <- is.na(cfg$files2export$start[pathes_to_gdx])
+      needfulldatagdx <- names(cfg$files2export$start[pathes_to_gdx][cfg$files2export$start[pathes_to_gdx] == cfg_main$title & !gdx_na])
+      message("In ", RData_file, ", use current fulldata.gdx path for ", paste(needfulldatagdx, collapse = ", "), ".")
+      cfg$files2export$start[needfulldatagdx] <- fulldatapath
+
+      save(cfg, file = RData_file)
+
+      # Subsequent runs will be started using submit.R, if all necessary gdx files were generated
+      gdx_exist <- grepl(".gdx", cfg$files2export$start[pathes_to_gdx])
+
+      if (all(gdx_exist | gdx_na)) {
+        message("Starting subsequent run ",run)
+        source("scripts/start/submit.R")
+        submit(cfg)
+        started_any_subsequent_run <- TRUE
+      } else {
+        message(run, " is still waiting for: ",
+        paste(unique(cfg$files2export$start[pathes_to_gdx][!(gdx_exist | gdx_na)]), collapse = ", "), ".")
+      }
+    } # end of loop through possible subsequent runs
 
     # Set cfg back to original
     cfg <- cfg_main
 
-    # 4. Create script file that can be used later to restart the subsequent runs manually.
+    # Create script file that can be used later to restart the subsequent runs manually.
     # In case there are no subsequent runs (or it's coupled runs), the file contains only
     # a small message.
 
     subseq_start_file  <- paste0(cfg$results_folder,"/start_subsequentruns_manually.R")
   
-    if(!any(cfg$RunsUsingTHISgdxAsInput$path_gdx_ref == cfg$title)) {
-      write("cat('\nNo subsequent run was set for this scenario\n')",file=subseq_start_file)
+    if(!any(cfg$RunsUsingTHISgdxAsInput == cfg$title)) {
```

Comment:
Can this comparison work, since cfg$RunsUsingTHISgdxAsInput is a data frame?

Reply:
Yes, I checked that carefully.